### PR TITLE
charts: add Dynamic Summary of selected benchmarks

### PR DESCRIPTION
**Pull request by terceiro**

### CODE REVIEW

**Comment by terceiro** [comment]
this includes the commits in #91; only the last 4 commits are new.

File: api/views.py [hunk 1]
```diff
@@ -166,6 +168,47 @@ def get_queryset(self):
...
+def dynamic_benchmark_summary(request):
+    branch = request.query_params.get('branch')
+    environment = request.query_params.get('environment')
+    benchmarks = request.query_params.getlist('benchmarks')
+
+    testjobs = benchmarks_models.TestJob.objects.filter(environment__identifier=environment)
+    testjobs = testjobs.select_related('result')
+    testjobs = testjobs.filter(result__branch_name=branch)
+    if settings.IGNORE_GERRIT is False:
+        testjobs = testjobs.filter(result__gerrit_change_number=None)
```
Comment (mwasilew) [inline api/views.py]: I think the meaning of IGNORE_GERRIT is to allow any values on gerrit related fields, not only None.

**Comment by mwasilew** [comment]
When I choose a category (summary), the main chart doesn't become a summary of summaries. Is this how it's suppose to work? Only single benchmarks seem to affect the main summary.

**Comment by terceiro** [comment]
@mwasilew IMO it doesn't make sense to take geomean of a geomean ... so yes this is how it is supposed to work. Maybe I should list explicitly (e.g. in the chart title) which benchmarks are  being used for the dynamic summary?

**Comment by mwasilew** [comment]
@terceiro +1 on the idea of listing the benchmarks. I agree there is little sense in geomean of geomeans. But the UI is a bit misleading. On the other hand it might get crowded if one adds a lot of measurements to the geomean chart. Maybe the list of metrics should be presented outside of the chart?

**Comment by terceiro** [comment]
I will think of something ;-)

**Comment by terceiro** [comment]
I think the less intrusive solution is to list the benchmarks used in the summary in a tooltip, like this:

tooltip

**Comment by mwasilew** [comment]
Looks good. Let me do a quick smoke test of this.

**Pull request merged**